src/models/Account.js

```diff
@@ -29,10 +27,10 @@ const AccountSchema = new Schema({
             return !this.isAdmin;
         },
         validate: {
-            validator: function(companyRef) {
-                return !!companyRef && !this.isAdmin;
+            validator: function(company) {
+                return !!company !== this.isAdmin;
             },
```

Comment:
Can we swap to arrow function and make this a one-liner?

```js
validator: company => !!company !== this.isAdmin;
```

Reply:
See the other comment